src/object/object_representation.c

```diff
@@ -1352,23 +1354,91 @@ or_put_varchar_internal (OR_BUF * buf, char *string, int charlen, int align)
   else
     {
       rc = or_put_byte (buf, 0xFF);
-
-      if (rc == NO_ERROR)
-	{
-	  OR_PUT_INT (&net_charlen, charlen);
-	  rc = or_put_data (buf, (char *) &net_charlen, OR_INT_SIZE);
-	}
+      compressable = 1;
     }
   if (rc != NO_ERROR)
     {
       return rc;
     }
 
-  /* store the string bytes */
-  rc = or_put_data (buf, string, charlen);
-  if (rc != NO_ERROR)
+  if (compressable == 1)
     {
-      return rc;
+      /* Alloc memory */
+      wrkmem = (lzo_voidp) malloc (LZO1X_1_MEM_COMPRESS);
```

Comment:
Future optimization : do not allocate wrkmem, but use a preallocated object (per thread_entry).